pkg/logql/ast.go

```diff
@@ -959,22 +962,38 @@ func (e *VectorAggregationExpr) Walk(f WalkFn) {
 	e.left.Walk(f)
 }
 
+type VectorMatching struct {
+	On      bool
+	Include []string
+}
+
 type BinOpOptions struct {
-	ReturnBool bool
+	ReturnBool     bool
+	VectorMatching *VectorMatching
 }
 
 type BinOpExpr struct {
 	SampleExpr
 	RHS  SampleExpr
 	op   string
-	opts BinOpOptions
+	opts *BinOpOptions
 }
 
 func (e *BinOpExpr) String() string {
-	if e.opts.ReturnBool {
-		return fmt.Sprintf("(%s %s bool %s)", e.SampleExpr.String(), e.op, e.RHS.String())
+	op := e.op
+	if e.opts != nil {
+		if e.opts.ReturnBool {
+			op = fmt.Sprintf("%s bool", op)
+		}
+		if e.opts.VectorMatching != nil {
+			if e.opts.VectorMatching.On {
```

Comment:
Is this covered by a tests ? I think we have stringer tests somewhere.

Reply:
yes, it is covered in parser test - https://github.com/garrettlish/loki/blob/onOrIgnoring/pkg/logql/parser_test.go#L2267
and engine test - https://github.com/garrettlish/loki/blob/onOrIgnoring/pkg/logql/engine_test.go#L639.